src/Gui/ViewProviderGeometryObject.cpp

```diff
@@ -77,9 +77,22 @@ const App::PropertyIntegerConstraint::Constraints intPercent = {0,100,1};
 ViewProviderGeometryObject::ViewProviderGeometryObject() : pcBoundSwitch(0),pcBoundColor(0)
 {
     ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/View");
-    unsigned long shcol = hGrp->GetUnsigned("DefaultShapeColor",3435973887UL); // light gray (204,204,204)
+    bool randomColor = hGrp->GetBool("RandomColor", true);

```

Comment:
This line means that if the RandomColor setting is non-existant, it will default to True, meaning everybody with an already installed FreeCAD will get random color without knowing what happened. I would prefer to set the default to false, so things will continue as they are for existing users